kore/src/Kore/Rewrite/SMT/Evaluator.hs

```diff
@@ -139,7 +142,7 @@ filterMultiOr multiOr = do
         Conditional variable term ->
         Simplifier (Maybe (Conditional variable term))
     refute p =
-        evalConditional p Nothing <&> \case
+        evalConditional ErrorInFilterMultiOr p Nothing <&> \case
```

Comment:
A bit unsure about this case... could `filterMultiOr` be used in the context of a simplification?
But not a blocker, even if this is the case, the new code still improves matters.

Reply:
Yeah, multi-or gets used in half a dozen other places. Wasn't sure how detailed we want to go. @ana-pantilie

Reply:
We can leave it like this for now.